lm_eval/utils.py

```diff
@@ -104,12 +149,51 @@ def complete_code(
             for sample, generated_tokens in zip(generated_tasks, generated_tokens):
                 gen_token_dict[sample].append(generated_tokens)
 
+    def strip_left_padding(tokens, pad_token_id):
+        """Strip left padding from tokens."""
+        for i, token in enumerate(tokens):
+            if token != pad_token_id:
+                return tokens[i:]
+        return tokens
```

Comment:
This implementation from DS1000 is probably more efficient
```suggestion
        num_non_pad = tokens.eq(pad_token_id).sum()
        return tokens[num_non_pad:]
```

Reply:
Actually, I'm going to roll this back. Introduces bug when padding appears elsewhere, e.g., as eos token. We only want to strip from left.